mp/src/game/shared/neo/neo_gamerules.cpp

```diff
@@ -1962,6 +1960,22 @@ void CNEORules::ClientDisconnected(edict_t* pClient)
 			ghost->SetRemoveable(false);
 			pNeoPlayer->Weapon_Detach(ghost);
 		}
+
+		// Save XP/death counts
+		if (neo_sv_player_restore.GetBool())
+		{
+			const CSteamID steamID = GetSteamIDForPlayerIndex(pNeoPlayer->entindex());
+			if (steamID.IsValid())
+			{
+				const uint32 accountID = steamID.GetAccountID();
+				const RestoreInfo restoreInfo{
+					.xp = pNeoPlayer->m_iXP.Get(),
+					.deaths = pNeoPlayer->DeathCount(),
+				};
+				m_pRestoredInfos.Remove(accountID); // Insert doesn't override so attempt remove first
+				m_pRestoredInfos.Insert(accountID, restoreInfo);
```

Comment:
Perhaps:
```cpp
bool didInsert;
auto idx = m_pRestoredInfos.Insert(accountID, restoreInfo, &didInsert);
if (!didInsert) // i.e. exists already
{
	m_pRestoredInfos[idx] = restoreInfo;
}
```
so that we don't need to remove data before each write.